src/app/services/query.service.ts

```diff
@@ -597,6 +597,162 @@ export class QueryService {
       );
   }
 
+  /**
+   * Import query from a file
+   * @param event
+   */
+  public importQuery(event) {
```

Comment:
I think the query import, step1 import and step2 import are different, i.e. the formats of the queries imported in these three places are different. I would put the import file logic into the respective component controllers, as they are DOM element interaction, and keep the query.service more about count update and manipulation of parsed queries.